### PR TITLE
Gremlin.net: Added a TimeSpanConverter for GraphSON.

### PR DESCRIPTION
**Pull request by danielcweber**
I can add more converters for extended types if appreciated and open the appropriate ticket in JIRA.

### CODE REVIEW

**Comment by spmallette** [comment]
Hello and thanks for contributing. I guess Gremlin.NET doesn't support all of extended IO yet:

**Comment by danielcweber** [comment]
Alright, I will try to follow the suggested process.

**Comment by FlorianHockmann** [comment]
> Not sure if @FlorianHockmann or @jorgebay have any further comments on how to proceed with this.

Not really, I think you already mentioned everything.

> I guess you would have to write your own body of tests for these serializers since the process suite doesn't cover them.

Having unit tests that don't require a Gremlin Server instance for most functionality is a good idea in general in my opinion as it makes it easier to find bugs. That's why we also have unit tests in Gremlin.Net for the serialization of all the types that are probably also covered by some of the process suite tests.

And also thanks from me for contributing to TinkerPop!

File: gremlin-dotnet/src/Gremlin.Net/Structure/IO/GraphSON/TimeSpanConverter.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+using Newtonsoft.Json.Linq;
+
+namespace Gremlin.Net.Structure.IO.GraphSON
+{
+    internal class TimeSpanConverter : IGraphSONSerializer, IGraphSONDeserializer
+    {
+        public Dictionary<string, dynamic> Dictify(dynamic objectData, GraphSONWriter writer)
+        {
+            TimeSpan value = objectData;
+            return GraphSONUtil.ToTypedValue("Duration", XmlConvert.ToString(value), "gx");
```
Comment (jorgebay) [inline gremlin-dotnet/src/Gremlin.Net/Structure/IO/GraphSON/TimeSpanConverter.cs]: `XmlConvert.ToString()` uses ISO-8601 representation (ie `PT8H6M12.345S`)?

I see no info on the MSDN: https://msdn.microsoft.com/en-us/library/z1955e3x

We are using `PTnHnMnS` as defined by java Duration: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toString--

Comment (danielcweber) [reply gremlin-dotnet/src/Gremlin.Net/Structure/IO/GraphSON/TimeSpanConverter.cs]: Seems to be implicit by how XML treats durations: https://www.w3.org/TR/xmlschema-2/#duration. The output of XmlConvert.ToString() follows the PTnHnMnS format.

Comment (jorgebay) [reply gremlin-dotnet/src/Gremlin.Net/Structure/IO/GraphSON/TimeSpanConverter.cs]: Oh, nice TIL!

**Comment by danielcweber** [comment]
-> https://github.com/apache/tinkerpop/pull/842

**Pull request closed without merging**